parser/unparser_formatting_options.go

```diff
@@ -0,0 +1,52 @@
+package parser
```

Comment:
Usually, it's good to define interfaces (or the functional option in this case) close to the API that consumes it -- unless you think this will grow to the point that it's a large component separate from the unparser (e.g. you would need to write tests specific to the apis in this file), you might consider just adding this to unparser.go

Reply:
I'll move it to unparser. There's a possibility that it might grow, but I think YAGNI applies at this point. We can always pull it out into its own file later on.